### PR TITLE
Added new types.

### PR DESCRIPTION
**Pull request by RazrFalcon**
Added new `pango` types:
- `Style`
- `Variant`
- `Weight`
- `Stretch`
- `LayoutIter`
- `Gravity`
- `GravityHint`
- `Script`
- `Direction`
- `BidiType`
- `FontMask`
- `LayoutLine`
- `EllipsizeMode`
- `WrapMode`

A new `Rectangle` implementation.

Reexported `PANGO_SCALE` constant.

Added new dependency: `bitflags`

### CODE REVIEW

File: src/auto/enums.rs [hunk 1]
```diff
@@ -40,3 +40,181 @@ impl FromGlib<ffi::PangoAlignment> for Alignment {
...
+    SemiExpanded,
+    Expanded,
+    ExtraExpanded,
+    UltraExpanded,
+    #[doc(hidden)]
+    __Nonexhaustive(()),
+}
+
+#[doc(hidden)]
+impl ToGlib for Stretch {
```
Comment (EPashkin) [inline src/auto/enums.rs]: I don't see removing `pub use ffi::PangoStretch as Stretch;` from `lib.rs` for this and others.
And this line override `pub use auto::*` and generating will be unusable to enduser.
Can you generate all reexported and remove reexports.

Comment (EPashkin) [inline src/auto/enums.rs]: Sorry, I looked to wrong commit, you already remove reexport,
but, please, remove others too.

File: src/auto/rectangle.rs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// This file was generated by gir (aacfc92) from gir-files (71d73f0)
+// DO NOT EDIT
+
+use ffi;
+use glib::translate::*;
```
Comment (EPashkin) [inline src/auto/rectangle.rs]: This file last from attempt to generate rectangle, remove it

Comment (RazrFalcon) [inline src/auto/rectangle.rs]: I can't prevent `gir` from generating it? Or I should just remove it from `git` and ignore it?

File: src/widgets/rectangle.rs [hunk 3]
```diff
@@ -2,32 +2,67 @@
...
+#[repr(C)]
+pub struct Rectangle {
+    pub x: i32,
+    pub y: i32,
+    pub width: i32,
+    pub height: i32,
 }
 
-impl Rectangle for PangoRectangle {
-    fn new(x: i32, y: i32, width: i32, height: i32) -> PangoRectangle {
```
Comment (EPashkin) [inline src/widgets/rectangle.rs]: Restore this function in `impl Rectangle {` may be someone used it.

File: src/auto/enums.rs (hunk 1 above)
Comment (RazrFalcon) [inline src/auto/enums.rs]: I hope it would be easy. I'vs relapsed only that types that I need.

File: src/widgets/rectangle.rs (hunk 3 above)
Comment (RazrFalcon) [inline src/widgets/rectangle.rs]: Ok.

**Comment by EPashkin** [comment]
Also better try generate more to uncomment almost all `Layout` and `LayoutIter` members.

File: src/auto/rectangle.rs (hunk 2 above)
Comment (EPashkin) [inline src/auto/rectangle.rs]: You don't removed it after generate attempt, gir don't regenerate it now.

**Comment by RazrFalcon** [comment]
About warnings:
```
warning: unused import: `glib_ffi`, #[warn(unused_imports)] on by default
 --> src/auto/enums.rs:5:5
  |
5 | use glib_ffi;
  |     ^^^^^^^^

warning: unused import: `glib::error::ErrorDomain`, #[warn(unused_imports)] on by default
 --> src/auto/enums.rs:6:5
  |
6 | use glib::error::ErrorDomain;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^
```
Should I just comment this lines? Or remove them completely.

**Comment by EPashkin** [comment]
As it in generated code, currently you only can ignore it.

File: src/widgets/rectangle.rs (hunk 3 above)
Comment (RazrFalcon) [inline src/widgets/rectangle.rs]: Done.

File: src/auto/enums.rs (hunk 1 above)
Comment (RazrFalcon) [inline src/auto/enums.rs]: Done.

File: src/auto/rectangle.rs (hunk 2 above)
Comment (RazrFalcon) [inline src/auto/rectangle.rs]: Done.

**Comment by GuillaumeGomez** [comment]
Seems good to me. Anything else remaining for you @EPashkin?

**Comment by EPashkin** [comment]
Generate
"Pango.LayoutLine",
"Pango.EllipsizeMode",
"Pango.WrapMode",

**Comment by RazrFalcon** [comment]
Done.

**Comment by EPashkin** [comment]
Seems you forgot add `auto/layout_line.rs`

**Comment by RazrFalcon** [comment]
D'oh.

**Comment by EPashkin** [comment]
Thanks. 👍

File: src/widgets/rectangle.rs [hunk 4]
```diff
@@ -2,17 +2,23 @@
...
-        PangoRectangle {
+impl Rectangle {
+    pub fn new(x: i32, y: i32, width: i32, height: i32) -> Rectangle {
+        Rectangle {
             x: x as c_int,
```
Comment (EPashkin) [inline src/widgets/rectangle.rs]: Hm, as c_int unneeded

Comment (RazrFalcon) [inline src/widgets/rectangle.rs]: It's old code. Not mine.

Comment (EPashkin) [inline src/widgets/rectangle.rs]: Old code created PangoRectangle with c_int fields, you add struct with i32 fields

Comment (RazrFalcon) [inline src/widgets/rectangle.rs]: Done.

**Comment by EPashkin** [comment]
Again 👍

**Comment by GuillaumeGomez** [comment]
I restart CI but otherwise it's all good. Thanks!

**Pull request merged**